crates/biome_js_analyze/src/lint/nursery/use_throw_new_error.rs

```diff
@@ -0,0 +1,180 @@
+    RuleSource,
+};
+use biome_console::markup;
+use biome_diagnostics::Applicability;
+use biome_js_factory::make;
+use biome_js_syntax::{
+    AnyJsExpression, JsCallExpression, JsNewExpression, JsParenthesizedExpression, JsSyntaxKind, T,
+};
+use biome_rowan::{AstNode, BatchMutationExt, TriviaPieceKind};
+
+use crate::JsRuleAction;
+
+declare_rule! {
+    /// Require `new` when throwing an error.
+    ///
+    /// While it's possible to create a new error without using the new keyword, it's better to be explicit.
+    ///
+    /// Rule matches errors when their name ends with the word "Error" and the first character is uppercase.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// throw Error();
+    /// ```
+    /// ```js,expect_diagnostic
+    /// throw TypeError('biome');
+    /// ```
+    /// ```js,expect_diagnostic
+    /// throw lib.TypeError();
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```js
+    /// throw new Error();
+    /// ```
+    /// ```js
+    /// throw new TypeError('biome');
+    /// ```
+    /// ```js
+    /// throw new lib.TypeError();
+    /// ```
+    ///
+    pub UseThrowNewError {
+        version: "next",
+        name: "useThrowNewError",
+        sources: &[RuleSource::EslintUnicorn("throw-new-error")],
+        recommended: false,
+        fix_kind: FixKind::Unsafe,
+    }
+}
+
+impl Rule for UseThrowNewError {
+    type Query = Ast<JsCallExpression>;
+    type State = ();
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let node = ctx.query();
+
+        if !is_in_throw_statement(node) {
+            return None;
+        }
+
+        let callee = &node.callee().ok()?.omit_parentheses();
+        let name = match callee {
+            AnyJsExpression::JsIdentifierExpression(ident_expr) => Some(
+                ident_expr
+                    .name()
+                    .ok()?
+                    .value_token()
+                    .ok()?
+                    .token_text_trimmed(),
+            ),
+            AnyJsExpression::JsStaticMemberExpression(member_expr) => Some(
+                member_expr
+                    .member()
+                    .ok()?
+                    .value_token()
+                    .ok()?
+                    .token_text_trimmed(),
+            ),
+            _ => None,
+        }?;
+
+        if name.ends_with("Error") && name.chars().next()?.is_uppercase() {
+            return Some(());
+        }
+
+        None
+    }
+
+    fn diagnostic(ctx: &RuleContext<Self>, _state: &Self::State) -> Option<RuleDiagnostic> {
+        let node = ctx.query();
+        Some(RuleDiagnostic::new(
+            rule_category!(),
+            node.range(),
+            markup! {
+                "Use "<Emphasis>"new"</Emphasis>" when throwing an error."
```

Comment:
Let's follow our rule pillars: https://biomejs.dev/linter/#rule-pillars

It seems `1.` and `2.` are missing

Reply:
I updated the diagnostic message, do you think I need to add note why it was matched? Something like line from documentation: Rule matches errors when their name ends with the word "Error" and the first character is uppercase.

Reply:
We should give a brief explanation of why we're suggesting `new Error()` as opposed to `Error()`.

Reply:
Add note "Instantiate Error with new keyword for consistency with modern builtins."